### PR TITLE
Fix Indeterminate State not Triggered from 'false' state

### PR DESCRIPTION
**Pull request by vincent-leonardo**
#76

Hi, after tracing the calls of indeterminate toggling I've found the source of this minor bug.
The problems happens with the default checkbox $formatters by angular.
#### The default checkbox formatters[0] on **angular.js:24100**

``` javascript
ctrl.$formatters.push(function(value) {
  return equals(value, trueValue);
});
```

So I did a quick hack to replace the default formatters in **initMaybe** @ **bsSwitch.js**
#### bsSwitch.js:153

``` javascript
controller.$formatters[0] = function(value) {
  if (value === undefined) {
    return value;
  }
  return angular.equals(value, getTrueValue());
}
```

### CODE REVIEW

**Comment by frapontillo** [comment]
Awesome! Can you please add a test for the false-to-indeterminate switch? Try to cover all edge cases if possible. Thanks!

**Comment by vincent-leonardo** [comment]
Done. I am not familiar with writing test case with karma *(in fact it's the first time I've written one)*
I don't use this library extensively, so I might not think of some edge cases.

Btw, I didn't realize indeterminate includes null value as well. Added it.

File: test/spec/directives/bsSwitchSpec.js [hunk 1]
```diff
@@ -723,4 +723,140 @@ describe('Directive: bsSwitch', function () {
...
+      scope.model = true;
+      scope.$apply();
+      expect(element.hasClass(CONST.SWITCH_INDETERMINATE_CLASS)).toBeFalsy();
+      expect(element.hasClass(CONST.SWITCH_ON_CLASS)).toBeTruthy();
+      expect(element.hasClass(CONST.SWITCH_OFF_CLASS)).toBeFalsy();
+      scope.model = null;
+      scope.$apply();
+      expect(element.hasClass(CONST.SWITCH_INDETERMINATE_CLASS)).toBeTruthy();
+      expect(element.hasClass(CONST.SWITCH_OFF_CLASS)).toBeFalsy();
+      expect(element.hasClass(CONST.SWITCH_ON_CLASS)).toBeTruthy();
```
Comment (frapontillo) [inline test/spec/directives/bsSwitchSpec.js]: If it is indeterminate, how come the on class is set?

**Comment by vincent-leonardo** [comment]
I thought that's how your library behaves?
All your test codes were written that way.

File: test/spec/directives/bsSwitchSpec.js (hunk 1 above)
Comment (vincent-leonardo) [inline test/spec/directives/bsSwitchSpec.js]: I thought that's how your library behaves?
All your test codes were written that way.

**Comment by SiobhanBaynes** [comment]
Hi guys, Any idea when this will be merged?

**Comment by ygilliot** [comment]
Since you ( @frapontillo ) validated, all checks have passed and branch has no conflicts, could you please merge to base?

I would prefer using "official" patched version than modifiying code file myself like a savage.
My thanks to @vincent-leonardo for resolving this bug.

**Comment by frapontillo** [comment]
@ygilliot I'm still not convinced by https://github.com/frapontillo/angular-bootstrap-switch/pull/121#r78273580

**Comment by frapontillo** [comment]
@vincent-leonardo can I have an update on this?

**Comment by vincent-leonardo** [comment]
@frapontillo as I mentioned before, your test cases were written such that in `indeterminate` state the class will still be `on`. You can check your original test spec [bsSwitchSpec.js:313](https://github.com/frapontillo/angular-bootstrap-switch/blob/develop/test/spec/directives/bsSwitchSpec.js#L313-L315)

**Comment by frapontillo** [comment]
Thanks @vincent-leonardo!

**Comment by vincent-leonardo** [comment]
@frapontillo no problem, glad it helps.
